### ISSUE TITLE
virt.get_disk output doesn't provide complete qemu-img data

### ISSUE THREAD

**Issue by cbosdo** [saltstack/salt#48085]
### Description of Issue/Question

Run `virt.get_disks` on a VM with such a setup:

* At least one disk with a qcow2 image with several backing files chained to each other
* At least one non-qcow2 image

The first problem is rather an inconsistency: the detailed `qemu-img info` output is only added for qcow2 files... why not other disk images like raw, qed, etc?

The output of the `qemu-img info` is rather badly parsed, for example:

* the snapshots date doesn't have the time part,
* if the snapshot name has a space in it, the name won't be complete
* the `format specific` lines are added to the snapshot list... they could be dropped or moved to the parent level

### Setup

This can be reproduced even with proper unit tests.

### Steps to Reproduce Issue

See description part

### Versions Report

All versions have it

**Comment by cbosdo** [comment]
The parsing code could be changed as follows to easily fix it:

* Use qemu-img  `--output json` parameter
* Use qemu-img `--backing-chain` to get the complete chain of backing files... not just the first one

However, the drawback is that the json output doesn't show the same keys for the different values. Along the same lines, the timestamps are just integers, not human-readble values. Would that be a serious problem for users to use these keys / values or should we map them to the current ones for compatibility?

Note, I am already working on fixing this, will submit a PR later

**Comment by cbosdo** [comment]
Output of `qemu-img info $PWD/test-suma-overlay.qcow2`

```
image: /var/lib/libvirt/images/test-suma-overlay.qcow2
file format: qcow2
virtual size: 24G (25769803776 bytes)
disk size: 216K
cluster_size: 65536
backing file: test-suma.qcow2 (actual path: /var/lib/libvirt/images/test-suma.qcow2)
Snapshot list:
ID        TAG                 VM SIZE                DATE       VM CLOCK
1         first-snap                0 2018-06-13 10:13:07   00:00:00.000
2         second-snap               0 2018-06-13 10:13:12   00:00:00.000
3         third snap                0 2018-06-13 11:32:10   00:00:00.000
Format specific information:
    compat: 1.1
    lazy refcounts: false
    refcount bits: 16
    corrupt: false
```

This even raises an error, as the code splits the snapshot columns using `\s` regular expression class. The corresponding output with `--output json --backing-chain` parameters

```json
[
    {
        "snapshots": [
            {
                "vm-clock-nsec": 0,
                "name": "first-snap",
                "date-sec": 1528877587,
                "date-nsec": 380589000,
                "vm-clock-sec": 0,
                "id": "1",
                "vm-state-size": 0
            },
            {
                "vm-clock-nsec": 0,
                "name": "second-snap",
                "date-sec": 1528877592,
                "date-nsec": 933509000,
                "vm-clock-sec": 0,
                "id": "2",
                "vm-state-size": 0
            },
            {
                "vm-clock-nsec": 0,
                "name": "third snap",
                "date-sec": 1528882330,
                "date-nsec": 188282000,
                "vm-clock-sec": 0,
                "id": "3",
                "vm-state-size": 0
            }
        ],
        "virtual-size": 25769803776,
        "filename": "/var/lib/libvirt/images/test-suma-overlay.qcow2",
        "cluster-size": 65536,
        "format": "qcow2",
        "actual-size": 221184,
        "format-specific": {
            "type": "qcow2",
            "data": {
                "compat": "1.1",
                "lazy-refcounts": false,
                "refcount-bits": 16,
                "corrupt": false
            }
        },
        "full-backing-filename": "/var/lib/libvirt/images/test-suma.qcow2",
        "backing-filename": "test-suma.qcow2",
        "dirty-flag": false
    },
    {
        "virtual-size": 25769803776,
        "filename": "/var/lib/libvirt/images/test-suma.qcow2",
        "cluster-size": 65536,
        "format": "qcow2",
        "actual-size": 393744384,
        "format-specific": {
            "type": "qcow2",
            "data": {
                "compat": "1.1",
                "lazy-refcounts": false,
                "refcount-bits": 16,
                "corrupt": false
            }
        },
        "dirty-flag": false
    }
]
```

The exception is due to the snapshot name having a space in it... without this snapshot, the output looks like this one:

```yaml
workstation.tf.local:
    ----------
    vda:
        ----------
        backing file:
            /var/lib/libvirt/images/test-suma.qcow2
        cluster_size:
            65536
        disk size:
            212K
        file:
            /var/lib/libvirt/images/test-suma-overlay.qcow2
        file format:
            qcow2
        image:
            /var/lib/libvirt/images/test-suma-overlay.qcow2
        snapshots:
            |_
              ----------
              date:
                  2018-06-13
              id:
                  1
              tag:
                  first-snap
              time:
                  10:13:07
              vmclock:
                  00:00:00.000
              vmsize:
                  0
            |_
              ----------
              date:
                  2018-06-13
              id:
                  2
              tag:
                  second-snap
              time:
                  10:13:12
              vmclock:
                  00:00:00.000
              vmsize:
                  0
            |_
              ----------
              id:
                  Format
              tag:
                  specific
              vmsize:
                  information:
            |_
              ----------
              id:
              tag:
                  compat:
              vmsize:
                  1.1
            |_
              ----------
              date:
                  false
              id:
              tag:
                  lazy
              vmsize:
                  refcounts:
            |_
              ----------
              date:
                  16
              id:
              tag:
                  refcount
              vmsize:
                  bits:
            |_
              ----------
              id:
              tag:
                  corrupt:
              vmsize:
                  false
        type:
            disk
        virtual size:
            24G (25769803776 bytes)
```

**Comment by cbosdo** [comment]
@gtmanfred, @cachedout any opinion on what to do with those changes? Should I use different properties as the values are presented differently or stick with these names?

**Comment by cbosdo** [comment]
@Ch3LL I wanted to finish deprecating libvirt:hypervisor to submit the PR... but yeah can submit earlier if needed. Maybe @thatch45 or @holmboe would have an opinion since they authored quite a lot of the virt things

**Comment by holmboe** [comment]
@cbosdo I think it is a good thing to retire `libvirt:hypervisor` setting as you have set out to do. :+1:

**Comment by cbosdo** [comment]
Fixed in 2019.2.0
